Package/Sublime Text Syntax Definition/Sublime Text Syntax Definition.sublime-syntax

```diff
@@ -330,6 +330,15 @@ contexts:
         - include: comment
         - include: yaml-tags-anchors
         - include: include
+    - match: \-(?=\s*{{plain_scalar_but_not_block_key}})
+      scope: punctuation.definition.block.sequence.yaml
+      push:
+        - meta_scope: meta.include-list.sublime-syntax meta.flow-sequence.yaml
+        - match: $\n?
```

Comment:
Capturing the newline will cause it to also have the meta scope applied, which seems weird to me considering the context won't span the next line.
Instead, only match on the anchor.

```suggestion
        - match: $
```

Alternatively, you could capture the indentation of the first item and only pop the context on a line with less indentation (whereas the `-` counts towards the indentation), but I don't think that's necessary.

Reply:
The `\n`  is captured because I saw context not to be suggested, if the cursor was at the end of the line. After a double check, I seem to be wrong.

While working with the change for a while, I found the normal `keywords` not to be part of the completions anymore, if the caret is at the first line of the sequence.

```
- match: test
  push:
    - m|  <-- no match suggested
```

Need to look into it as well.

Reply:
Use the "Alternative solution" with a dedicated context.